bootstrap/system_dependency.py

```diff
@@ -0,0 +1,55 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import Dict, Set
+
+from minifi_option import MinifiOptions
+from package_manager import PackageManager
+
+
+def _create_system_dependencies(minifi_options: MinifiOptions) -> Dict[str, Set[str]]:
+    system_dependencies = {'patch': {'patch'}, 'make': {'make'}}
+    if minifi_options.is_enabled("ENABLE_EXPRESSION_LANGUAGE"):
+        system_dependencies['bison'] = {'bison'}
+        system_dependencies['flex'] = {'flex'}
+    if minifi_options.is_enabled("ENABLE_LIBARCHIVE"):
+        system_dependencies['libarchive'] = {'libarchive'}
+    if minifi_options.is_enabled("ENABLE_PCAP"):
+        system_dependencies['libpcap'] = {'libpcap'}
+    if minifi_options.is_enabled("ENABLE_USB_CAMERA"):
+        system_dependencies['libusb'] = {'libusb'}
+        system_dependencies['libpng'] = {'libpng'}
+    if minifi_options.is_enabled("ENABLE_GPS"):
+        system_dependencies['gpsd'] = {'gpsd'}
+    if minifi_options.is_enabled("ENABLE_COAP"):
+        system_dependencies['automake'] = {'automake'}
+        system_dependencies['autoconf'] = {'autoconf'}
+        system_dependencies['libtool'] = {'libtool'}
+    if minifi_options.is_enabled("ENABLE_LUA_SCRIPTING"):
+        system_dependencies['lua'] = {'lua'}
+    if minifi_options.is_enabled("ENABLE_PYTHON_SCRIPTING"):
+        system_dependencies['python'] = {'python'}
+    if minifi_options.is_enabled("MINIFI_OPENSSL"):
+        system_dependencies['openssl'] = {'perl'}
```

Comment:
OpenSSL also needs NASM, at least on Windows. Is it handled elsewhere?

Reply:
Thats packaged with strawberryperl, (with the whole toolchain but that causes other problems). Usually we wipe the c:\Strawberry\c\bin dir to fix those problems, but then we also wipe the NASM which is required, so we had to install that aswell, instead Im deleting the installed toolchain but leave NASM alone. 
https://github.com/martinzink/nifi-minifi-cpp/blob/python_bootstrap/bootstrap/package_manager.py#L178

(installing NASM separatly came with its own issues, since neither choco nor winget sets the PATH for NASM)

Reply:
Wiping the Strawberry Perl bin directory is a bit too extreme IMO, possibly ruining the user's existing installation. Would it be possible to only fix the PATH envvar instead? Like maybe moving the Strawberry Perl bin directory to the end of it, so the normal patch command is found first.

Reply:
I replaced this approach with only temporarly changing the path and not deleting anything.
(I've also added NASM as a separate dependency)
https://github.com/apache/nifi-minifi-cpp/pull/1681/commits/e611b1e44e82a01e082c3b8a1099d2d7591d7401#diff-0ebcdc25be3be8a77e8b404eb6962891695836a8ca6bfa65607a2622d114f35dR235-R237